src/Native/jitinterface/jithost.cpp

```diff
@@ -6,9 +6,36 @@
 
 #include "dllexport.h"
 
+class JitConfigProvider
+{
+public:
+    virtual int getIntConfigValue(
+        const wchar_t* name, 
+        int defaultValue
+        ) = 0;
+
+    virtual int getStringConfigValue(
+        const wchar_t* name,
+        wchar_t* retBuffer,
+        int retBufferLength
+        ) = 0;
+};
+
+// Native implementation of the JIT host.
+// The native implementation calls into JitConfigProvider (implemented on the managed side) to get the actual
+// configuration values.
+// This dance is necessary because RyuJIT calls into the JitHost as part of the process shutdown (to free up
+// strings). JitHost therefore can't be implemented in managed code (because managed runtime might have
```

Comment:
Yes ... the JIT shutdown is a mess ... there is a good chance it will get cleaned up because of it is causing other issues as well.